src/main/haskell/kore/src/Kore/Step/BaseStep.hs

```diff
@@ -176,6 +172,15 @@ instance
     freshVariableWith (ConfigurationVariable a) n =
         ConfigurationVariable $ freshVariableWith a n
 
+instance
+    Unparse (variable level) =>
+    Unparse (StepperVariable variable level)
+  where
+    unparse =
+        \case
+            AxiomVariable var -> unparse var
+            ConfigurationVariable var -> unparse var
```

Comment:
Unparsing these the same way makes me somewhat nervous. It will probably work in most cases (almost all?), but in the rare case when we have two distinct variables unparsing to exactly the same thing, reparsing the resulting string will produce a pattern with a completely different meaning; also, debugging output will be confusing.

Prefixing the variables somehow would make the problem less likely, and would probably make the debug output more readable.